src/alerts/management/commands/create_alerts_from_csv_file.py

```diff
@@ -0,0 +1,105 @@
+    pipenv run python manage.py create_alerts_from_csv_file file/path/name.csv --alert_querystring 'perimeter=70971-nouvelle-aquitaine' --alert_date_created '2020-08-20' # noqa
+    pipenv run python manage.py create_alerts_from_csv_file file/path/name.csv --alert_querystring 'perimeter=70971-nouvelle-aquitaine' --alert_date_created '2020-08-20 2:00' --alert_frequency weekly # noqa
+    pipenv run python manage.py create_alerts_from_csv_file file/path/name.csv --alert_querystring 'perimeter=70971-nouvelle-aquitaine'  --alert_validated # noqa
+
+    CSV File example:
+    email
+    test@email.com
+    test2@email.com
+    """
+
+    def add_arguments(self, parser):
+        parser.add_argument('csv_file', nargs=1, type=str)
+        parser.add_argument(
+            '--csv_delimiter', nargs='?', type=str,
+            default=DEFAULT_CSV_DELIMITER
+        )
+        parser.add_argument(
+            '--alert_querystring', nargs='?', type=str,
+            default=DEFAULT_ALERT_QUERYSTRING
+        )
+        parser.add_argument(
+            '--alert_title', nargs='?', type=str,
+            default=DEFAULT_ALERT_TITLE
+        )
+        parser.add_argument(
+            '--alert_date_created', type=datetime.datetime.fromisoformat,
+            default=timezone.now()
+        )
+        parser.add_argument(
+            '--alert_frequency', nargs='?', type=str,
+            choices=Alert.FREQUENCIES, default=DEFAULT_ALERT_FREQUENCY
+        )
+        parser.add_argument(
+            '--alert_validated', action='store_true',
+            default=DEFAULT_ALERT_VALIDATED
+        )
+
+    def handle(self, *args, **options):
+        # csv file
+        csv_path = os.path.abspath(options['csv_file'][0])
+        csv_delimiter = options['csv_delimiter']
+        # alert details
+        alert_querystring = options['alert_querystring']
+        alert_title = options['alert_title']
+        alert_date_created = options['alert_date_created']
+        alert_frequency = options['alert_frequency']
+        alert_validated = options['alert_validated']
+
+        with open(csv_path) as csv_file:
+            csvreader = csv.DictReader(csv_file, delimiter=csv_delimiter)
+            for index, row in enumerate(csvreader):
+                self.create_alert(row['email'], alert_querystring, alert_title, alert_date_created, alert_frequency, alert_validated) # noqa
+
+    def create_alert(self, alert_email, alert_querystring, alert_title, alert_date_created, alert_frequency, alert_validated): # noqa
+        print(alert_date_created)
+        alert = Alert.objects.create(
+            email=alert_email,
+            querystring=alert_querystring,
+            title=alert_title,
+            alert_frequency=alert_frequency,
+            date_created=alert_date_created,
+        )
+
+        self.stdout.write('Alert {} created for {}'.format(
+            alert.title, alert.email))
+
+        # send confirmation email
+        if alert_validated:
```

Comment:
Je n'ai pas suivi l'origine de la demande ni la discussion, mais j'ai quelques questions sur cette validation.

Le RGPD nous contraint à recueillir et conserver la trace d'un consentement explicite avant de stocker et utiliser des adresses individuelles.

Le fait que les adresses soient opt-in chez le partenaire ne lui donne pas le droit de les transférer chez nous sauf si cela a été consenti explicitement. De même, ça ne nous donne pas le droit de les récupérer pour les utiliser à notre guise.

Est-ce que vous vous êtes assurés d'être bien dans la légalité ? Si c'est le cas, avez-vous correctement documenté l'origine du consentement ?